c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/C2Heartbeat.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Apache NiFi - MiNiFi
+ * Copyright 2014-2018 The Apache Software Foundation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.api;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import java.io.Serializable;
+import javax.xml.bind.annotation.XmlTransient;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+/**
+ * TODO add Builder interface
+ */
+@ApiModel
+public class C2Heartbeat implements Serializable {
+    private static final long serialVersionUID = -1168790962754919591L;
+
+    // Internal, not part of REST API
+    private String identifier;
+    private Long created;
```

Comment:
Is there a reason for using `Long` as opposed to something like `Instant` or `OffsetDateTime`?

Reply:
I think only because we know it will be serialized since it's part of the protocol. If using date/time classes is more readable/maintainable, I can use those instead.

Reply:
Thanks for the explanation, that seems reasonable.  Using a Long will allow for up to millisecond precision, and if that aligns with other implementations, that sounds good.